### DISCUSSION TITLE
Unrelated changes in lockfile when updating a dependency version?`

### DISCUSSION THREAD

**Discussion by codepunkt** [pnpm/pnpm#6547] [Q&A]
Using `pnpm@8.5.0` in a workspaces monorepo, when running `pnpm i @wat/wat@VERSION` in a workspace to upgrade the `@wat/wat` dependency of that specific workspace to a given version, the lockfile has a lot of changes that seem unrelated. All of them happen in parantheses in the values of `version` keys, e.g. `@swc/core` changing it's version. Our `resolution-mode` is set to highest, if that helps.

Why do these happen? Is there a way to prevent them?

![image](https://github.com/pnpm/pnpm/assets/603683/73db3371-8aff-4423-813c-78473a24cd3c)

**Comment by zkochan** [comment]
Probably it shouldn't happen, unless `@swc/core` is a dependency of `@wat/wat`.
